internal/resource/config/serversettings/logsettings/server_settings_log_settings_data_source.go

```diff
@@ -0,0 +1,128 @@
+func NewServerSettingsLogSettingsDataSource() datasource.DataSource {
+	return &serverSettingsLogSettingsDataSource{}
+}
+
+// serverSettingsLogSettingsDataSource is the datasource implementation.
+type serverSettingsLogSettingsDataSource struct {
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+type serverSettingsLogSettingsDataSourceModel struct {
+	Id            types.String `tfsdk:"id"`
+	LogCategories types.Set    `tfsdk:"log_categories"`
+}
+
+// GetSchema defines the schema for the datasource.
+func (r *serverSettingsLogSettingsDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	schema := schema.Schema{
+		Description: "Describes the settings related to server logging.",
+		Attributes: map[string]schema.Attribute{
+			"log_categories": schema.SetNestedAttribute{
+				Description: "The log categories defined for the system and whether they are enabled.",
+				Computed:    true,
+				Optional:    false,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"id": schema.StringAttribute{
+							Description: "The ID of the log category. This field must match one of the category IDs defined in log4j-categories.xml.",
+							Computed:    true,
+							Optional:    false,
+						},
+						"name": schema.StringAttribute{
+							Description: "The description of the log category.",
+							Computed:    true,
+							Optional:    false,
+						},
+						"description": schema.StringAttribute{
+							Description: "The description of the log category.",
+							Computed:    true,
+							Optional:    false,
+						},
+						"enabled": schema.BoolAttribute{
+							Description: "Determines whether or not the log category is enabled. The default is false.",
+							Computed:    true,
+							Optional:    false,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	id.ToSchema(&schema)
+	resp.Schema = schema
+}
+
+// Metadata returns the datasource type name.
+func (r *serverSettingsLogSettingsDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_server_settings_log_settings"
+}
+
+func (r *serverSettingsLogSettingsDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, _ *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+
+func readServerSettingsLogSettingsDataSource(ctx context.Context, r *client.LogSettings, state *serverSettingsLogSettingsDataSourceModel) diag.Diagnostics {
+	var diags, respDiags diag.Diagnostics
+	state.Id = types.StringValue("server_settings_log_settings_id")
+
+	logCategorySlice, respDiags := types.SetValueFrom(ctx, types.ObjectType{AttrTypes: logCategoriesAttrTypes}, r.LogCategories)
```

Comment:
You can just put state.LogCategories on the left here, instead of reassigning below.
`state.LogCategories, respDiags = types.SetValueFrom(...)`